content/en/docs/tasks/configure-pod-container/security-context.md

```diff
@@ -385,6 +385,12 @@ omit the `CAP_` portion of the constant.
 For example, to add `CAP_SYS_TIME`, include `SYS_TIME` in your list of capabilities.
 {{< /note >}}
 
+### Limitations of Linux Capabilities for Non-Root Containers
+
+When running containers as non-root users, it's important to be aware that Linux capabilities might not be fully granted due to the security model of Linux. This limitation can prevent non-root containers from performing operations that require specific capabilities, even if those capabilities are explicitly granted in the container's security context.
```

Comment:
Hi,

I think it would be worth mentioning some examples of this behavior and make the wording more precise to avoid confusion.

In a PR [comment](https://github.com/kubernetes/website/pull/45150#issuecomment-1947920793), you mention [AppArmor](https://github.com/kubernetes/kubernetes/issues/108247) as an example. LSMs are indeed one example of a security mechanism that imposes additional restrictive rules about what a process can and cannot do. An LSM blocking an operation, however, is not exactly the same as the Linux capabilities not being granted. Capabilities are being granted, but certain operations might be  disallowed by other parts of the Linux security stack.

It's also not clear to me whether the workaround of setting capabilities directly on the binary (as mentioned below) is a valid workaround since I'd expect that LSMs, with proper configuration, could provide more restrictive controls for that case as well.